src/els_formatting_provider.erl

```diff
@@ -0,0 +1,115 @@
+-module(els_formatting_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , is_enabled_document/0
+        , is_enabled_range/0
+        , is_enabled_on_type/0
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+
+%% Keep the behaviour happy
+-spec is_enabled() -> boolean().
+is_enabled() -> is_enabled_document().
```

Comment:
Would it make sense for this to be an `orelse` between the three modes?

Reply:
Each of these flags is used to set a specific capability value back to the client during initialisation, so they need to be separate.